src/core/server/plugins/plugins_service.ts

```diff
@@ -86,6 +86,9 @@ export interface PluginsServiceDiscoverDeps {
   environment: InternalEnvironmentServicePreboot;
 }
 
+const OSS_PATH_REGEX = /[\/|\\]src[\/|\\]plugins[\/|\\]/; // Matches src/plugins directory on POSIX and Windows
+const XPACK_PATH_REGEX = /[\/|\\]x-pack[\/|\\]plugins[\/|\\]/; // Matches x-pack/plugins directory on POSIX and Windows
```

Comment:
wow, quite a complex regexp. cannot we just test that a path contains `x-pack` string?

Reply:
This just searches for `/x-pack/plugins/` or `\\x-pack\\plugins\\`. I wanted to include "plugins" too because searching just for "src" is not good enough. So if we are searching for "/src/plugins/", why not "/x-pack/plugins/"?

You should have seen the original regexp I was testing 😅 
```
[^\/|\\]+[\/|\\]x-pack[\/|\\]plugins[\/|\\][^\/|\\]+
```
(That's to check that there is at least one directory level above and one directory level below)

Anyway, this passed my personal bar of "good enough" string matching. If you think it needs to be changed though, I'm happy to do so.

Reply:
> I wanted to include "plugins" too because searching just for "src" is not good enough.

Why so? We know that a plugin from `src` directory can be placed under `src/plugins` path only.

Reply:
The plugin's path is the absolute file path on the system. So if we just searched for "src" in the path, and someone placed the whole Kibana directory in another folder with "src" in its name, then every plugin would be detected as an OSS plugin.

Searching for `/src/plugins/` and `/x-pack/plugins/` seems like a reasonable effort at reducing false positives without getting too complicated, I hope you would agree.

Reply:
> The plugin's path is the absolute file path on the system.  

ok, that changes a lot